src/WebJobs.Script/Metrics/IHostMetricsCollector.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Script.Metrics
+{
+    public interface IHostMetricsCollector
```

Comment:
Usually a "collector" has Add methods that allow data to be pumped in. This interface seems more like a "provider"? Is this pattern you're using with IHostMetricsCollector and HostMetricsMeter an established pattern for Otel or something? I'm not familiar.

Reply:
No real reason I picked collector, so I'm happy to rename this to provider if that makes most sense.

As for the HostMetricsMeter -> IHostMetricsCollector pattern: using a `Meter` with a `MeterListener` is the recommended .NET Metrics pattern, I didn't end up naming it HostMetricsListener since it was doing more than just listening for metrics, the class is collecting metrics over time and making them available to consumers. The pattern can be found here: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/metrics-collection#create-a-custom-collection-tool-using-the-net-meterlistener-api.